app/models/concerns/hyrax/collection_behavior.rb

```diff
@@ -78,8 +89,39 @@ def member_object_ids
       ActiveFedora::Base.search_with_conditions("member_of_collection_ids_ssim:#{id}").map(&:id)
     end
 
+    protected
+
+      def collection_type=(collection_type)
+        @collection_type = collection_type
```

Comment:
I believe this should raise a `Runtime` exception if the given `collection_type` is not a Hyrax::CollectionType? Or an `ActiveRecord::AssociationTypeMismatch` error? It appears you could move the `collection_type_valid?` method logic into this method.